content/docs/command-reference/commit.md

```diff
@@ -17,65 +16,54 @@ positional arguments:
-
-- Sometimes we want to edit source code, config, or data files in a way that
-  doesn't cause changes in the results of their data pipeline. We might write
-  add code comments, change indentation, remove some debugging printouts, or any
-  other change that doesn't cause changed stage outputs. However, DVC will
-  notice that some <abbr>dependencies</abbr> have changed, and expect you to
-  reproduce the whole pipeline. If you're sure no pipeline results would change,
-  use `dvc commit` to force update the `dvc.lock` or `.dvc` files and cache.
-
-- In some cases, we have previously executed a stage, and later notice that some
-  of the files/directories used by the stage as dependencies or created as
-  outputs are missing from `dvc.yaml`. It is possible to
-  [add missing data to an existing stage](/docs/user-guide/how-to/add-deps-or-outs-to-a-stage),
-  and then `dvc commit` can be used to save outputs to the cache (and update
-  `dvc.lock`)
-
-- It's always possible to manually execute the command or source code used in a
-  stage without DVC (outputs must be unprotected or removed first in certain
-  cases, see `dvc unprotect`). Once the desired result is reached, use
-  `dvc commit` to update the `dvc.lock` file(s) and store changed data to the
-  cache.
-
-Let's take a look at what is happening in the first scenario closely. Normally
-DVC commands like `dvc add`, `dvc repro` or `dvc run` commit the data to the
-<abbr>cache</abbr> after creating or updating a `dvc.lock` or `.dvc` file. What
-_commit_ means is that DVC:
-
-- Computes a hash for the file/directory.
-- Enters the hash value and file name in the `dvc.lock` or `.dvc` file.
-- Tells Git to ignore the file/directory (adding them to `.gitignore`). (Note
-  that if the <abbr>project</abbr> was initialized with no Git support
-  (`dvc init --no-scm`), this does not happen.)
-- Adds the file(s) in question to the cache.
-
-There are many cases where the last step is not desirable (for example rapid
-iterations on an experiment). The `--no-commit` option prevents it (on the
-commands where it's available). The file hash is still computed and added to the
-`dvc.lock` or `.dvc` file, but the actual data is not cached. And this is where
-the `dvc commit` command comes into play: It performs that last step when
-needed.
-
-Note that it's best to avoid the last three scenarios. They essentially
-force-update the `dvc.lock` or `.dvc` files and save data to cache. They are
-still useful, but keep in mind that DVC can't guarantee reproducibility in those
-cases.
+<abbr>Caches</abbr> the current contents of files and directories tracked by
+DVC, and updates `dvc.lock` or `.dvc` files as needed.
+
+This can be useful for several scenarios, when the project is under development,
+or to force DVC to accept changed data (avoiding `dvc add` or `dvc repro`).
+We'll expand on these uses below.
+
+Normally, `dvc repro` and `dvc run` finish up with the same steps as `dvc add`,
```

Comment:
also, import? import-url?

generalize this?

Reply:
okay, actually they don not cache I guess (by default at least) ... I would try to generalize this somehow 

`Normally, commands that involve data tracking (dvc run, dvc repro, dvc add, etc) ...` ?

Reply:
> actually [import commands] don not cache I guess (by default

They never cached, I think, even if the `store: true` option gets merged (which BTW would change the concept of "remote storage" a little as it could contain files that the cache can never contain... But I digress).

This also made me realize that the explanation of `import/get --no-exec` was wrong with regards to caching — prob. because it was copied from `run`. Updated that in ce8ccad2

Reply:
> would try to generalize this... commands that involve data tracking...

But it's only run and repro, I think. And `add` only does the last part of what run/repro do: save the file hash and cache the file. So if it's just those 3, why say "etc." ?

Reply:
> But it's only run and repro

p.s. meaning, the ones that have a `--no-commit` option to skip that last "adding" part.

Reply:
`dvc commit <out>.dvc can be used if the data hashes are needed in the .dvc file, and to save existing data to the cache.` - this comes from the `dvc import`

cache != remote, import/import-url cache artifacts as far as I remember, after all

> But it's only run and repro, I think. And add only does the last part of what run/repro do: save the file hash and cache the file. So if it's just those 3, why say "etc." ?

hmm, what is the first part?

Reply:
> import/import-url cache artifacts as far as I remember

Yeah I may be confused. Let em double check on that... ⌛ 

> what is the first part?

"the last part" refers to the part of repro and run that has to do with the pipeline or stage structure and command (write or read dvc.yaml, determine what to execute, run commands, etc.). But I guess the whole "Normally, dvc repro and dvc run finish up..." paragraph and bullet list bellow can be simplified to avoid explaining too much about those commands here... ⌛

Reply:
> import/import-url cache artifacts... Let me double check

Yeah I was confused 🤦 — I reverted those changes (and noticed other problems in those refs but will leave that for later.)

Reply:
> the whole "Normally, dvc repro and dvc run finish up..." paragraph and bullet list bellow can be simplified

OK I generalized and simplified this without completely removing the bullet list (which I think makes the reading easier/more interesting).